### DISCUSSION TITLE
Doesn't return error message

### DISCUSSION THREAD

**Discussion by kelvin-zhao** [honojs/hono#2000] [Q&A]
Hi guys,

I'm using

`return c.text( '${ err.message }', 500 )`

to return an error message, but I don't see the message being returned, just a default internal error.
I did a console.log on my api, it does have an error message though.
How should I properly send the message back?

**Comment by linhub15** [comment]
My guess is that some code before that line is throwing the error, so by default `500` error will be returned to the browser with the internal error.

**Comment by kelvin-zhao** [comment]
The way I'm doing my code block is as follows, does the throw error in my try block doesn't hit the catch but sends the error back instantly?

```
app.post( '/', async ( c ) => {
	try {
		const { pin, campaign } = await c.req.json() as Payload

		If ( pin === 'xxx' ) {
			const openai = new OpenAI( { apiKey: c.env.API_KEY } )
			const result = await runOpenAI( openai, campaign, c.env.GPT_ID )

			return c.json( {
				'prediction': result.prediction,
				'summary': result.summary,
				'improvement': result.improvement,
			} )
		} else {
			throw new Error( 'wrong pin' )
		}
	} catch ( err ) {
		console.log( err.message )
		return c.text( `${ err.message }`, 500 )
	}
} )
```

My console.log was hit and I did get the message in my logs.

**Reply by linhub15** [reply]
Your code looks correct. Everything in the `try` block will be caught in your example above and returned as text the the client.

If you still see "Internal Server Error" it will be caused outside the `try` block.
I would need to see the stack trace in your console to see where it's coming from.

Here's a working example of returning the error message, based on your example.
https://dash.deno.com/playground/sad-rabbit-86

**Reply by kelvin-zhao** [reply]
Weirdly I removed everything on my end and leave the throw of wrong pin,
I see it being logged and still internal server error is sent back.
I'm so confused.

```
wrong pin
[wrangler:inf] POST / 500 Internal Server Error (3ms)
```

Is it the way I am taking in the response?

```
const res = await fetch( mode, {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
  },
    body: JSON.stringify( { pin: pin.value, campaign: campaign.value } ),
  } )
if ( !res.ok ) throw new Error( res.statusText )
```
		
When I check on res, there's no properties in there that logged the 'wrong pin' message.
Is there something else I should check on for the response text?

**Reply by linhub15** [reply]
Hard to dig any deeper without access to your code and the cloud environment settings...

Here's a few things I suggest you try to further debug this
- try running Hono locally without Wranger to find out if it's Hono, or Wrangler or your edge function provider
- remove all middleware, and then add them in one at a time to rule out if they are the cause
